src/passes/SafeHeap.cpp

```diff
@@ -100,11 +101,31 @@ struct AccessInstrumenter : public WalkerPass<PostWalker<AccessInstrumenter>> {
   }
 };
 
-struct FindDirectCallees : public WalkerPass<PostWalker<FindDirectCallees>> {
-public:
-  void visitCall(Call* curr) { callees.insert(curr->target); }
-  std::set<Name> callees;
-};
+static std::set<Name> findCalledFunctions(Module* module, Name startFunc) {
+  std::set<Name> called;
+  std::vector<Name> toVisit;
+
+  auto addFunction = [&](Name name) {
+    if (called.count(name) == 0) {
+      called.insert(name);
+      toVisit.push_back(name);
+    }
```

Comment:
I am pretty sure @tlively would like you to do `if (called.insert(name).second) toVisit.push...`